lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
I disagree. Unprotected critical sections are never "fine as they are". The deeper problem we're seeing is only one manifestation of possible things that can go wrong. So, protecting the critical section is clearly a "fix".